### PR TITLE
Tech centre trainee acceptance

### PR DESCRIPTION
**Pull request by MrFletcha**
Here we go:
- Reliance of Trainee Course types removed and given to Trainee Centre (Only did this because I thought it made sense in terms of class responsibility, if not then I can just change it back no worries)
- Enums added to Trainee Centre for reasons above
- The original set method has been deleted from Trainee for the centre course enum setting as this is now done in Trainee Centre
- New method added to Trainee centre to get and set value of Tech Centre specialism
- A test version of the set Centre Specialism is currently being used so I could define what the centres specialism is, but the one above is the random version we will use in the actual program.
- The above getter has been added to Centre interface so it can be used across factory classes 
- Trainee manager unplaced Trainees array changed to public purely so I could use it in tests, so feel free to change back
- Trainee Centre also has changes to add Trainee to check if the course type matches the tech centre type before adding, this does check for a null value first, ensuring that if the centre has no specialism then we add the trainee anyway. The difference is that if the tech centre we are adding to is not of the same type then the trainee is not added, but this does not dictate where that trainee goes after being rejected.
- I think that's all, sorry for rambling. Any questions then hit me up

### CODE REVIEW

File: src/main/java/com/sparta/spartaSimulator/model/WaitingList.java [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 public class WaitingList {
 
-    private static ArrayList<Trainee> waitingList = new ArrayList<>();
+    //made this public for tests in TechCentreTestsClass
+    public static ArrayList<Trainee> waitingList = new ArrayList<>();
```
Comment (NJohnston94) [inline src/main/java/com/sparta/spartaSimulator/model/WaitingList.java]: needs to be reset to private and accessed through getter

**Review by NJohnston94 — COMMENTED** [review]
WaitingList needs to be reset to private from public, otherwise all good

File: src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java [hunk 2]
```diff
@@ -46,8 +66,20 @@ public void checkCentreStatus() {
...
+        {
+            specialism = getCentreSpecialism().toString();
+        } catch (NullPointerException ex)
+        {
+            allTrainees.add(trainee);
+        }
+        if(trainee.getTraineeCourse().toString().equals(specialism))
+        {
+            allTrainees.add(trainee);
+        }
```
Comment (Danielall2) [inline src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java:82]: Does not look for other random trainees with the specified course. If the techCentre wants 13 trainees and there are 13 trainees in the waiting list with the required course, no guarantee they will be sent to centre, as there is no condition to pick another if they have the wrong course.

File: src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java [hunk 3]
```diff
@@ -2,20 +2,40 @@
...
+    private CentreSpecialism centreSpecialism;
+    private static final Random RANDOM = new Random();
 
     public enum CentreStatus{
         FULL,
         NEARLY_FULL,
         NOT_FULL
     }
 
+    public enum CentreSpecialism {
```
Comment (Iworkreallyhard) [inline src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java:24]: This shouldn't be in trainee centre. This should be in tech centre

File: src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java [hunk 4]
```diff
@@ -2,20 +2,40 @@
...
+import java.util.Arrays;
 import java.util.HashSet;
+import java.util.Random;
 
 public abstract class TraineeCentre {
 
     private HashSet<Trainee> allTrainees = new HashSet<>();
     private int MAX_CAPACITY;
     private CentreStatus centreStatus;
+    private CentreSpecialism centreSpecialism;
```
Comment (Iworkreallyhard) [inline src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java:15]: should be in tech centre

**Review by RashawnKH — COMMENTED** [review]
Question about if only the tech centres are being checked for course type, or if every centre is being checked. As only the tech centre has a specific course type

File: src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java [hunk 5]
```diff
@@ -72,4 +104,18 @@ public int getMaxCapacity() {
         return allTrainees;
     }
 
+    public CentreSpecialism getCentreSpecialism() {
```
Comment (Iworkreallyhard) [inline src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java:107]: tech centre

**Review by Iworkreallyhard — COMMENTED** [review]
could move centre specialism to tech centre from CentreManager

**Pull request closed without merging**